### PR TITLE
Fix mono/OSX build

### PR DESCRIPTION
**Pull request by radical**
- Disable FEATURE_PIPE_SECURITY for mono builds
- #if out a PipeSecurity use
- s,FEATURE_APPDOMAIN,FEATURE_BINARY_SERIALIZATION,
    for using INodePacketTranslator.TranslateDotNet

Issue #531

### CODE REVIEW

File: src/XMakeBuildEngine/BackEnd/Node/NodeConfiguration.cs [hunk 1]
```diff
@@ -142,7 +142,7 @@ public void Translate(INodePacketTranslator translator)
             translator.Translate(ref _nodeId);
             translator.Translate(ref _buildParameters, BuildParameters.FactoryForDeserialization);
             translator.TranslateArray(ref _forwardingLoggers, LoggerDescription.FactoryForTranslation);
-#if FEATURE_APPDOMAIN
+#if FEATURE_BINARY_SERIALIZATION
```
Comment (dsplaisted) [inline src/XMakeBuildEngine/BackEnd/Node/NodeConfiguration.cs]: What's the reason for this change?  You shouldn't be serializing AppDomainSetup unless AppDomain's are available.  Combining both also seems reasonable:

    #if FEATURE_APPDOMAIN && FEATURE_BINARY_SERIALIZATION

Comment (radical) [inline src/XMakeBuildEngine/BackEnd/Node/NodeConfiguration.cs]: I based this solely on the availability of `INodePacketTranslator.TranslateDotNet`, which is defined for `FEATURE_BINARY_SERIALIZATION` .

Comment (radical) [inline src/XMakeBuildEngine/BackEnd/Node/NodeConfiguration.cs]: And you are correct, _appDomainSetup is available only for FEATURE_APPDOMAIN, so, it should be a combination.

**Comment by radical** [comment]
I had committed a similar "fix" for `INodePacketTranslator.TranslateDotNet` use earlier, fixing that again in #535 .

**Comment by dsplaisted** [comment]
LGTM, though I would just include the one-line change from #535 in this PR instead of keeping it separate.

**Comment by radical** [comment]
Updated.

**Comment by AndyGerlicher** [comment]
LGTM
